File: Projects/Project_Cinema Website/backend/src/main/java/com/cinema_website/backend/service/FoodService.java
package com.cinema_website.backend.service;

import com.cinema_website.backend.dto.FoodDTO;
import com.cinema_website.backend.repository.FoodRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class FoodService {

    private final FoodRepository foodRepository;

    @Autowired
    public FoodService(FoodRepository foodRepository) {
        this.foodRepository = foodRepository;
    }

    // GET
    public FoodDTO getFood(int foodId) {
        return foodRepository.getFoodById(foodId);
    }

    public List<FoodDTO> getAllFoods() {
        return foodRepository.getAllFoods();
    }

    public List<String> getAllCategories() {
        return foodRepository.getAllCategories();
    }


    // INSERT
    public boolean addFood(FoodDTO food) {
        return foodRepository.addFood(food);
    }

    // UPDATE
    public boolean updateFood(int foodId, FoodDTO updatedFood) {
        return foodRepository.updateFoodById(foodId, updatedFood);
    }

    // DELETE
    public boolean deleteFood(int foodId) {
        return foodRepository.deleteFoodById(foodId);
    }
}

File: Projects/Project_Cinema Website/frontend/src/services/FoodService.js
import axios from "axios";

const baseURL = 'http://localhost:8080/api/foods'; // Replace with your actual API endpoint

class FoodService {

    // GET
    static async getFoodById(foodId) {
        return (await axios.get(`${baseURL}/getFood/${foodId}`)).data;
    }

    static async getAllFoods() {
        return (await axios.get(`${baseURL}/getAllFoods`)).data;
    }

    static async getAllCategories() {
        return (await axios.get(`${baseURL}/getAllCategories`)).data;
    }

    // POST
    static async addFood(newFood) {
        return (await axios.post(`${baseURL}/addFood`, newFood)).data;
    }

    // PUT (Update)
    static async updateFoodById(foodId, updatedFood) {
        return (await axios.put(`${baseURL}/updateFood/${foodId}`, updatedFood)).data;
    }

    // DELETE
    static async deleteFoodById(foodId) {
        return (await axios.delete(`${baseURL}/deleteFood/${foodId}`)).data;
    }
}

export default FoodService;
